mono/mini/exceptions-amd64.c

```diff
@@ -455,6 +466,7 @@ get_throw_trampoline (MonoTrampInfo **info, gboolean rethrow, gboolean corlib, g
 	mono_arch_flush_icache (start, code - start);
 
 	g_assert ((code - start) < kMaxCodeSize);
+	g_assert_checked (mono_arch_unwindinfo_validate_size (unwind_ops, MONO_MAX_TRAMPOLINE_UNWINDINFO_SIZE));
 
```

Comment:
I put it into a checked assert since the mono_arch_unwindinfo_validate_size does some additional work to walk through the list of unwind ops and classify them (the same that is done later when installing it into the unwind tables) accordingly to calculate size needed to serialize them. This work is only needed to validate the pre allocated buffer that in turn builds on a size assumption done in code. It is not depending on any input data, so as long as  the code doesn't change (adding new or change unwind ops used by windows) it will always be true. The performance impact will be in direct relationship with how many trampolines that gets created, so if this is not that frequently called in relation to everything else, we could make it a normal g_assert, but then we will do the calculation twice in release builds where it most likely should be true 100% of the times. I think this is something you normally would like to get noticed during development and fixed and then don't paid the additional cost in release builds. If the checked build assert (something we will run on windows debug builds during development) is a problem, I can switch to a normal g_assert. Thoughts?